src/vm/scanning.rs

```diff
@@ -4,33 +4,67 @@ use crate::util::ObjectReference;
 use crate::util::OpaquePointer;
 use crate::vm::VMBinding;
 
+/// VM-specific methods for scanning roots/objects.
 pub trait Scanning<VM: VMBinding> {
-    /// Scan stack roots after all mutators are paused
+    /// Scan stack roots after all mutators are paused.
     const SCAN_MUTATORS_IN_SAFEPOINT: bool = true;
-    /// Scan all the mutators within a single work packet
+
+    /// Scan all the mutators within a single work packet.
     ///
     /// `SCAN_MUTATORS_IN_SAFEPOINT` should also be enabled
     const SINGLE_THREAD_MUTATOR_SCANNING: bool = true;
+
+    /// Delegated scanning of a object, processing each pointer field
+    /// encountered. This method probably will be removed in the future,
+    /// in favor of bulk scanning `scan_objects`.
+    ///
+    /// Arguments:
+    /// * `trace`: The `TransitiveClosure` to use for scanning.
+    /// * `object`: The object to be scanned.
+    /// * `tls`: The GC worker thread that is doing this tracing.
     fn scan_object<T: TransitiveClosure>(
         trace: &mut T,
         object: ObjectReference,
         tls: OpaquePointer,
     );
+
+    #[deprecated]
     fn reset_thread_counter();
+
+    /// MMTk callls this method at the first time during a collection that thread's stacks
```

Comment:
This is not clear to me.
Does it mean?:
MMTk calls this method when the thread's stacks scanning is completed for the first time during a collection.

(completion may happen multiple times during a collection and this function is called on the first occurrence?)

Reply:
Yeah.